tests/v2/validation/networking/connectivity/port_test.go

```diff
@@ -0,0 +1,266 @@
+		assert.NoError(p.T(), err)
+		nodeIP := kubeapinodes.GetNodeIP(newNode, corev1.NodeInternalIP)
+
+		p.T().Log("Executing the SSH shell command on the node")
+		log, err := curlCommand(p.client, p.cluster.ID, fmt.Sprintf("%s:%s/name.html", nodeIP, strconv.Itoa(hostPort)))
+		require.NoError(p.T(), err)
+		require.True(p.T(), strings.Contains(log, daemonsetName))
+	}
+}
+
+func (p *PortTestSuite) TestNodePort() {
+	subSession := p.session.NewSession()
+	defer subSession.Cleanup()
+
+	log.Info("Creating new project and namespace")
+	_, namespace, err := projectsapi.CreateProjectAndNamespace(p.client, p.cluster.ID)
+	require.NoError(p.T(), err)
+
+	steveClient, err := p.client.Steve.ProxyDownstream(p.cluster.ID)
+	require.NoError(p.T(), err)
+
+	testContainerPodTemplate := newPodTemplateWithTestContainer()
+
+	daemonsetName := namegen.AppendRandomString("test-daemonset")
+
+	p.T().Logf("Creating a daemonset with the test container with name [%v]", daemonsetName)
+	daemonsetTemplate := shepworkloads.NewDaemonSetTemplate(daemonsetName, namespace.Name, testContainerPodTemplate, true, nil)
+	createdDaemonSet, err := steveClient.SteveType(workloads.DaemonsetSteveType).Create(daemonsetTemplate)
+	require.NoError(p.T(), err)
+	assert.Equal(p.T(), createdDaemonSet.Name, daemonsetName)
+
+	p.T().Logf("Waiting for daemonset [%v] to have expected number of available replicas", daemonsetName)
+	err = charts.WatchAndWaitDaemonSets(p.client, p.cluster.ID, namespace.Name, metav1.ListOptions{})
+	require.NoError(p.T(), err)
+
+	//It will allocate a port from a range 30000-32767
+	//Using a random port to avoid 'port in use' failures and allow the test to be rerun
+	nodePort := rand.IntN(2767) + 30000
+
+	serviceName := namegen.AppendRandomString("test-service")
+	p.T().Logf("Creating service with name [%v]", serviceName)
+	ports := []corev1.ServicePort{
+		{
+			Protocol: corev1.ProtocolTCP,
+			Port:     80,
+			NodePort: int32(nodePort),
+		},
+	}
+	nodePortservice := services.NewServiceTemplate(serviceName, namespace.Name, corev1.ServiceTypeNodePort, ports, daemonsetTemplate.Spec.Template.Labels)
+	serviceResp, err := services.CreateService(steveClient, nodePortservice)
+	require.NoError(p.T(), err)
+
+	err = services.VerifyService(steveClient, serviceResp)
+	require.NoError(p.T(), err)
+
+	p.T().Logf("Getting the node using the cluster name [%v]", p.clusterName)
+	query, err := url.ParseQuery(fmt.Sprintf("%s=%s", clusterNameAnnotation, p.clusterName))
+	assert.NoError(p.T(), err)
+
+	nodeList, err := steveClient.SteveType("node").List(query)
+	assert.NoError(p.T(), err)
+	assert.NotEmpty(p.T(), nodeList, err)
+	assert.NotEmpty(p.T(), nodeList.Data)
+
+	for _, machine := range nodeList.Data {
+		p.T().Log("Getting the node IP")
+		newNode := &corev1.Node{}
+		err = steveV1.ConvertToK8sType(machine.JSONResp, newNode)
+		assert.NoError(p.T(), err)
+
+		// Project Network Isolation should be enabled when setting up the cluster for this test
```

Comment:
can you add this to the readme, and probably add validation before the pod is setup that fails if PNI isn't set?

Reply:
I can add it on the README

Reply:
Ticket created https://github.com/rancher/qa-tasks/issues/1607